### PR TITLE
Audit logs pagination

### PR DESCRIPTION
**Pull request by MonikaSharma06**
Added pagination in Audit Logs with some changes in pagination component

Note: Need some more changes in pagination component

### CODE REVIEW

File: src/frontend/src/app/assets/assets.component.ts [hunk 1]
```diff
@@ -50,6 +50,23 @@ export class AssetsComponent implements OnInit {
   }
 
   /**
+   *  Go to the previous page
```
Comment (praveen-garg) [inline src/frontend/src/app/assets/assets.component.ts]: wrong comment

File: src/frontend/src/app/assets/assets.component.ts [hunk 2]
```diff
@@ -50,6 +50,23 @@ export class AssetsComponent implements OnInit {
...
+   */
+  onFirst(): void {
+    this.page = 1;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the next page
```
Comment (praveen-garg) [inline src/frontend/src/app/assets/assets.component.ts]: wrong comment

File: src/frontend/src/app/assets/assets.component.ts [hunk 3]
```diff
@@ -50,6 +50,23 @@ export class AssetsComponent implements OnInit {
...
+   */
+  onLast(n: number): void {
+    const p = Math.ceil(this.recordCount / this.limit) || 0;
+    this.page = p;
+    this.setLimitOffset();
+  }
+
+  /**
    *  Set limit and offset (it is internally called by goToPage(), onNext(), onPrev() methods)
```
Comment (praveen-garg) [inline src/frontend/src/app/assets/assets.component.ts]: and now for onFirst and  onLast

File: src/frontend/src/app/audit-log/audit-log.component.html [hunk 4]
```diff
@@ -57,5 +57,18 @@
       </tr>
       </tbody>
     </table>
+      <div *ngIf= filterdData> 
+        <my-pagination
+          [count]= this.recordCount
+          [page]="page"
+          [perPage]= this.limit
+          [pagesToShow]="3"
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.html]: 3 = first + middle + last ?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 5]
```diff
@@ -11,11 +11,15 @@ export class AuditLogComponent implements OnInit {
   public logSeverityList = [];
   public filterdData = [];
 
-  public limit: Number = 0;
-  public offset: Number = 0;
+  limit = 20;
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts:15]: why now its 20 ?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 6]
```diff
@@ -11,11 +11,15 @@ export class AuditLogComponent implements OnInit {
...
+  offset = 0;
   public source: String = '';
   public severity: String = '';
 
+  page = 1;             // Default page is 1 in pagination
+  recordCount = 100;    // Total no. of records during pagination (For now it is hard coded, TODO: FOGL-714)
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts]: for asset records too? can not we just have this for audit log?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 7]
```diff
@@ -24,6 +28,65 @@ export class AuditLogComponent implements OnInit {
...
+  /**
+   *  Go to the next page
+   */
+  onNext(): void {
+    this.page++;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the previous page
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts]: previous or first?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 8]
```diff
@@ -24,6 +28,65 @@ export class AuditLogComponent implements OnInit {
...
+   */
+  onFirst(): void {
+    this.page = 1;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Go to the next page
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts]: next or last?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 9]
```diff
@@ -24,6 +28,65 @@ export class AuditLogComponent implements OnInit {
...
+  /**
+   *  Go to the previous page
+   */
+  onPrev(): void {
+    this.page--;
+    this.setLimitOffset();
+  }
+
+  /**
+   *  Set limit and offset (it is internally called by goToPage(), onNext(), onPrev() methods)
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts]: is it duplicate method?

File: src/frontend/src/app/audit-log/audit-log.component.ts [hunk 10]
```diff
@@ -91,7 +164,11 @@ export class AuditLogComponent implements OnInit {
   }
 
   auditLogSubscriber() {
-    this.auditService.getAuditLogs(this.limit, this.offset, this.source, this.severity).
+    console.log('Limit: ', this.limit);
+    console.log('offset: ', this.offset);
+    console.log('tempOffset: ', this.tempOffset);
+    console.log('recordCount: ', this.recordCount);
+    this.auditService.getAuditLogs(this.limit, this.tempOffset, this.source, this.severity).
```
Comment (praveen-garg) [inline src/frontend/src/app/audit-log/audit-log.component.ts]: make above log one (or max 2 liner) e.g. `console.log('Limit: ', this.limit, 'Offset: ', this.offset);`

**Review by praveen-garg — CHANGES_REQUESTED** [review]
@monika-Dianomic  please get reviewed this  with Shariq src/frontend/src/app/pagination/pagination.component.html

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 5 above)
Comment (monika-Dianomic) [reply src/frontend/src/app/audit-log/audit-log.component.ts]: It is 20.

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 6 above)
Comment (MonikaSharma06) [reply src/frontend/src/app/audit-log/audit-log.component.ts]: But it is in audit log, not in asset records. (File name is- audit-log.component.ts)

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 9 above)
Comment (MonikaSharma06) [reply src/frontend/src/app/audit-log/audit-log.component.ts]: Not duplicate method in the same file (It is in different files). That method was in `assets.component.ts` and it is in `audit-log.component.ts`

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 8 above)
Comment (MonikaSharma06) [reply src/frontend/src/app/audit-log/audit-log.component.ts]: It is next.

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 7 above)
Comment (MonikaSharma06) [reply src/frontend/src/app/audit-log/audit-log.component.ts]: previous

File: src/frontend/src/app/audit-log/audit-log.component.ts (hunk 5 above)
Comment (MonikaSharma06) [reply src/frontend/src/app/audit-log/audit-log.component.ts:15]: By default, limit is 20 for pagination.

**Comment by praveen-garg** [comment]
closing in favor of https://github.com/praveen-garg/FogLAMP/pull/48

**Pull request closed without merging**